### ISSUE TITLE
Upgrade from Version 1.4.0 --> 1.4.2 introduces rb-gsl dependency, downgrades spreadsheet

### ISSUE THREAD

**Issue by rdlugosz** [SciRuby/statsample#28]
_[As mentioned in my [issue in the old repo](https://github.com/clbustos/statsample/issues/43#issuecomment-95295541)]…_

It appears that going from `1.4.0` to `1.4.2` creates some unexpected dependency changes that should not be a part of a Patch-level release:

1. The `rb-gsl` gem is now required. This is problematic on Heroku where it cannot be easily built.
1. The `spreadsheet` gem gets pinned to an older Minor release; on my app this mandated a _downgrade_ of `spreadsheet` from `0.9.9` to `0.6.9`. Not sure if this is intentional or matters, but it was surprising if nothing else.
1. If the `minimization` gem is not pinned to `0.2.1` a similar dependency is created with the bump to `0.2.2`.  I have opened [an issue over there](https://github.com/SciRuby/minimization/issues/8).

Here's a peek at the diff of Gemfile.lock when `statsample ~> 1.4.0` and `mimization = 0.2.1`:
```
-    statsample (1.4.0)
-      dirty-memoize (~> 0.0)
+    statsample (1.4.2)
+      awesome_print
+      dirty-memoize
       distribution
-      extendmatrix (~> 0.3.1)
-      fastercsv (> 0)
-      minimization (~> 0.2.0)
+      extendmatrix
+      minimization
+      rb-gsl
       reportbuilder (~> 1.4)
       rserve-client
-      rubyvis
-      spreadsheet (~> 0.6)
-      statsample-bivariate-extension (> 0)
-    statsample-bivariate-extension (1.1.0)
-      distribution (~> 0.6)
+      rubyvis (~> 0.5.0)
+      spreadsheet (~> 0.6.5)
```

**Comment by agarie** [comment]
In commit [8ce01](https://github.com/SciRuby/minimization/commit/8ce01c55116a6e5ca10abe8d76ad71daae7ea16d) I removed the dependency on rb-gsl and pushed a new version (1.4.3) to rubygems.

I also updated Minimization. Everything should be working right now. :P

Thank you for these issues!

**Comment by rdlugosz** [comment]
Very cool, thx. Any thoughts on why the `spreadsheet` gem is being pinned to the `0.6` version? More of a curiosity, but previously we were at `0.9.9` and their most current release is `1.0.3`.

**Comment by agarie** [comment]
Not really. I refactored `statsample` to use a gemspec instead of `hoe` and the `rb-gsl` dep was introduced by mistake -- I was using it anyway on my system and Travis already installed it. Thankfully you and other folks complained and it's fixed now. I had that version of `spreadsheet` installed here and it didn't break anything, so I pinned it.

I opened #29 to work on this. Nothing is broken right now, but leaving it outdated doesn't seem like a good idea...
